### PR TITLE
Fix issue #70 and issue #88

### PR DESCRIPTION
**Pull request by hishamco**
I made this PR to fix the issues #70 and #88

### CODE REVIEW

**Comment by Eilon** [comment]
@hishamco is this the full fix for #70? I thought the fix for that was quite different, no?

@kirthik can you please review? If this looks good, please resolve #88.

**Comment by hishamco** [comment]
@Eilon regarding the issue #70 the original bug causes when the ```DefaultRequestCulture``` is not supported, this PR is fix this issue after the change made by @kirthik at the commit https://github.com/aspnet/Localization/commit/7b53bdd43e2e605a8eb686f22ceb558f46a1c4f5, i think we should create a new issue for another related scenarios that discussed to #70.
Please let me know if you have another suggestion @Eilon ?

**Comment by Eilon** [comment]
Why not resolve issue #70 with a full fix for the scenario described there? What would the new issue say that isn't already said in #70?

The fix here seems fine for #88.

**Comment by hishamco** [comment]
Yep it's fine for issue #88, but if you have a look to the first post in the issue #70 uncomment the following line
```csharp
//, DefaultRequestCulture = new RequestCulture(new CultureInfo("nl-NL"), new CultureInfo("nl-NL"))
```
getting the excepting result, which is done by this PR after the change in the commit https://github.com/aspnet/Localization/commit/7b53bdd43e2e605a8eb686f22ceb558f46a1c4f5, other posts raised other issues related to the original bug. i hope you understand what I mean

File: src/Microsoft.AspNet.Localization/AcceptLanguageHeaderRequestCultureProvider.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ public override Task<RequestCulture> DetermineRequestCulture([NotNull] HttpConte
 
                         requestCulture = ValidateRequestCulture(requestCulture);
 
-                        if (requestCulture.Culture == culture)
+                        if (requestCulture != null && requestCulture.Culture == culture)
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Localization/AcceptLanguageHeaderRequestCultureProvider.cs]: `requestCulture?.Culture == culture`

Comment (hishamco) [inline src/Microsoft.AspNet.Localization/AcceptLanguageHeaderRequestCultureProvider.cs]: Looks good :smile: , thanks @davidfowl

**Comment by kirthik** [comment]
I don't understand how this change is touching #70. Can you please edit PR title so it says it just fixes #88?

File: test/Microsoft.AspNet.Localization.Tests/AccessLanguageHeaderRequestCultureProviderTest.cs [hunk 2]
```diff
@@ -106,5 +106,38 @@ public class AccessLanguageHeaderRequestCultureProviderTest
                 var response = await client.GetAsync(string.Empty);
             }
         }
+        
+        [Fact]
+        public async void OmitDefaultRequestCultureShouldNotThrowNullReferenceException_And_ShouldGetTheRightCulture()
```
Comment (kirthik) [inline test/Microsoft.AspNet.Localization.Tests/AccessLanguageHeaderRequestCultureProviderTest.cs]: How is this test different from `GetFallbackLanguage_ReturnsFirstNonNullCultureFromSupportedCultureList`. Except for the added `SupportedUICultures` list in this test, I don't see any difference.

**Comment by hishamco** [comment]
@kirthik the main difference between ```OmitDefaultRequestCultureShouldNotThrowNullReferenceException_And_ShouldGetTheRightCulture``` and ```GetFallbackLanguage_ReturnsFirstNonNullCultureFromSupportedCultureList``` is https://github.com/aspnet/Localization/blob/dev/test/Microsoft.AspNet.Localization.Tests/AccessLanguageHeaderRequestCultureProviderTest.cs#L55

File: test/Microsoft.AspNet.Localization.Tests/AccessLanguageHeaderRequestCultureProviderTest.cs (hunk 2 above)
Comment (hishamco) [inline test/Microsoft.AspNet.Localization.Tests/AccessLanguageHeaderRequestCultureProviderTest.cs]: @kirthik the main difference between OmitDefaultRequestCultureShouldNotThrowNullReferenceException_And_ShouldGetTheRightCulture and GetFallbackLanguage_ReturnsFirstNonNullCultureFromSupportedCultureList is https://github.com/aspnet/Localization/blob/dev/test/Microsoft.AspNet.Localization.Tests/AccessLanguageHeaderRequestCultureProviderTest.cs#L55

Comment (kirthik) [inline test/Microsoft.AspNet.Localization.Tests/AccessLanguageHeaderRequestCultureProviderTest.cs]: The line you pointed to is `GetFallbackLanguage_ReturnsFromSupportedCulture_AcceptLanguageListContainsSupportedCultures`. I am talking about `GetFallbackLanguage_ReturnsFirstNonNullCultureFromSupportedCultureList`

**Comment by hishamco** [comment]
@kirthik the main issue in #70 that when we omit the ```DefaultRequestCulture``` the ```CurrentCulture``` & ```CurrentUICulture``` will be **es-US**, this PR fix it, i know there are some scenarios related to issue #70, if you see the rename of the title is a proper choice i will do it now

File: test/Microsoft.AspNet.Localization.Tests/AccessLanguageHeaderRequestCultureProviderTest.cs (hunk 2 above)
Comment (hishamco) [inline test/Microsoft.AspNet.Localization.Tests/AccessLanguageHeaderRequestCultureProviderTest.cs]: Oops .. sorry, seems introducing ```SupportedUICultures``` property cause the error
```csharp
SupportedUICultures = new List<CultureInfo>
{
    new CultureInfo("ar-YE")
}
```
Can you check please ..

Comment (hishamco) [inline test/Microsoft.AspNet.Localization.Tests/AccessLanguageHeaderRequestCultureProviderTest.cs]: In fact support both ```SupportedCultures``` & ```SupportedUICultures``` cause the error, I tested in my end. If you notice not ```OmitDefaultRequestCultureShouldNotThrowNullReferenceException_And_ShouldGetTheRightCulture``` is the only test that support both of them

**Comment by kirthik** [comment]
merged 61106d36a17239d60ccd4cb6e0257a267d9f32ba

**Pull request closed without merging**